### PR TITLE
Remove DeviceFarm references

### PR DESCRIPTION
**Pull request by OdNairy**
## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -58,13 +58,7 @@ To run device tests there are few options:
...
-2. Trigger via CI by adding `[run device tests]` to a git commit message. These tests also run "nightly". 
-(Examples can also be run as tests on CI by adding `[run app device tests]`.)
-
-3. Trigger tests on AWS Device Farm from the command line by running `make test-with-device-farm SCHEME=MapboxTestHost APP_NAME=MapboxTestHost`. This requires certain environment variables to be set; please see the makefile for these.
-
-4. Trigger tests on a local device (connected by USB) using the same setup as Device Farm testing by running:
-`make local-test-with-device-farm-ipa SCHEME=MapboxTestHost CONFIGURATION=Release ENABLE_CODE_SIGNING=1`
+2. Trigger via CI by manual approve of `optional-tests` workflow. These tests also run "nightly".
 
 ## Integration Tests
```
Comment (evil159) [inline DEVELOPING.md:64]: There is still a reference to the AWS Device Farm on line 65(just below this one), not sure what's the scope of this removal, but should it be replaced as well with Firebase Test Lab?

File: Makefile [hunk 2]
```diff
@@ -134,12 +130,6 @@ build-app-for-simulator:
...
 # Devices - SDK
 
-USB_DEVICE_ID := $(strip $(shell system_profiler SPUSBDataType | sed -n -E -e "/(iPhone|iPad)/,/Serial/s/ *Serial Number: *(.+)/\1/p" ))
-
-ifneq ($(USB_DEVICE_ID),)
-USB_DEVICE_DESTINATION := -destination 'platform=ios,id=$(USB_DEVICE_ID)'
-endif
-
 # Xcode build command for building for device. The CODE_SIGNING_* variables are so that no code signing occurs on CI -
 # this is because AWS Device Farm re-signs the applications. This may need to change if a different provider is used.
```
Comment (evil159) [inline Makefile:134]: AWS Device Farm mentioned here as well, as well as on line 198(if that's even a problem).

**Review by macdrevx — COMMENTED** [review]
No further comments beyond what @evil159 mentioned.

File: Makefile (hunk 2 above)
Comment (OdNairy) [reply Makefile:134]: I'd like to keep it as it describes the exact behaviour of Device Farm. I'm not sure are we safe to remove all code signing part for now as we now do codesign for Test Lab.

**Review by evil159 — APPROVED** [review]

**Pull request merged**